app/features/receive/receive-cashu-token-hooks.tsx

```diff
@@ -0,0 +1,354 @@
+function isSameAccount(a: CashuAccount, b: CashuAccount) {
+  return a.mintUrl === b.mintUrl && a.currency === b.currency;
+}
+
+const getDefaultReceiveAccount = (
+  selectableAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  isCrossMintSwapDisabled: boolean,
+  defaultAccount: Account,
+): CashuAccount => {
+  const targetAccount =
+    isCrossMintSwapDisabled || defaultAccount.type !== 'cashu'
+      ? sourceAccount
+      : defaultAccount;
+
+  const matchingAccount = selectableAccounts.find((account) =>
+    isSameAccount(account, targetAccount),
+  );
+
+  // Fall back to source account if no match found
+  return matchingAccount ?? sourceAccount;
+};
+
+const getBadges = (
+  account: CashuAccount,
+  allAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  defaultAccount: Account,
+) => {
+  const badges: string[] = [];
+
+  if (sourceAccount.isTestMint) {
+    badges.push('Test Mint');
+  }
+  if (account.mintUrl === sourceAccount.mintUrl) {
+    badges.push('Source');
+  }
+  if (
+    defaultAccount.type === 'cashu' &&
+    isSameAccount(account, defaultAccount)
+  ) {
+    badges.push('Default');
+  }
+  if (!allAccounts.some((a) => a.mintUrl === account.mintUrl)) {
+    badges.push('Unknown');
+  }
+
+  return badges;
+};
+
+/**
+ * Returns a list of accounts that the user can select to receive the token.
+ * If cross mint swaps are disabled because we can't make a lightning payment
+ * from the source account, we only show the source account. Otherwise, we show
+ * the source account and all other non-testnet cashu accounts.
+ */
+const getSelectableAccounts = (
+  sourceAccount: CashuAccount,
+  isCrossMintSwapDisabled: boolean,
+  accounts: CashuAccount[],
+) => {
+  return isCrossMintSwapDisabled
+    ? [sourceAccount]
+    : [
+        sourceAccount,
+        ...accounts.filter(
+          (account) =>
+            !account.isTestMint && account.mintUrl !== sourceAccount.mintUrl,
+        ),
+      ];
+};
+
+/**
+ * Lets the user select an account to receive the token and returns data about the
+ * selectable accounts based on the source account and the user's accounts in the database.
+ */
+export function useReceiveCashuTokenAccounts(
+  sourceAccount: ExtendedCashuAccount,
+) {
+  const { data: accounts } = useAccounts({ type: 'cashu' });
+  const defaultAccount = useDefaultAccount();
+
+  const isCrossMintSwapDisabled = sourceAccount.isTestMint;
+  const selectableAccounts = getSelectableAccounts(
+    sourceAccount,
+    isCrossMintSwapDisabled,
+    accounts,
+  );
+  const defaultReceiveAccount = getDefaultReceiveAccount(
+    selectableAccounts,
+    sourceAccount,
+    isCrossMintSwapDisabled,
+    defaultAccount,
+  );
+
+  const [receiveAccountId, setReceiveAccountId] = useState<string>(
+    defaultReceiveAccount.id,
+  );
+  const receiveAccount = useAccount<ExtendedCashuAccount>(
+    receiveAccountId,
+    () => sourceAccount,
+  );
+
+  const accountWithBadges = selectableAccounts.map((a) => ({
```

Comment:
can we augment the selectable accounts with badges inside the getSelectableAccounts hook?